83 integrate twitter auth

andrii-zaharovskyi opened the pull request:
Cose #83

andrii-zaharovskyi committed:
Authorization via the Twitter provider has been added.

Authorization and code formatting

andrii-zaharovskyi committed:
The method AuthenticateGoogleFacebookUser has been renamed.

Rename and code formatting

andrii-zaharovskyi committed:
Squashed commit of the following:

commit a748f93b483e0f739c4a4e974be3fa7c9bdec6e8
Merge: bf7ad2f 6d92079
Author: Andrii Zaharovskyi <andrew.zaharovskyi@gmail.com>
Date:   Wed Sep 30 17:13:43 2020 +0300

    Merge pull request #105 from EventsExpress-SoftServe/sand0-patch-1

    Delete EventsExpress.xml

commit 6d92079c2ef419a4a81d453bf48ab0f42502a82a
Author: Alexandr Savchuk <44262349+sand0@users.noreply.github.com>
Date:   Wed Sep 30 17:08:24 2020 +0300

    Delete EventsExpress.xml

commit bf7ad2f4a2173022e87d69137ff05325a77b257e
Merge: 9a5beac da74169
Author: Yevhen Vasylynchuk <45979893+Avalon412@users.noreply.github.com>
Date:   Wed Sep 30 14:46:53 2020 +0300

    Merge pull request #104 from EventsExpress-SoftServe/99-fix-CommunaTimeAccoutning

    Fixed time accounting in 'Comunna'

commit da7416931aa28fe917ede872af447c428834ca81
Merge: a4f5073 9a5beac
Author: Yevhen Vasylynchuk <45979893+Avalon412@users.noreply.github.com>
Date:   Tue Sep 29 18:43:58 2020 +0300

    Merge branch 'development' into 99-fix-CommunaTimeAccoutning

commit a4f50734bca8a087fddf29057d23b397181f5c3e
Author: Женя Василинчук <bensonatg412@gmail.com>
Date:   Tue Sep 29 18:31:54 2020 +0300

    Fixed time accounting in 'Comunna'

iuriikhrystiuk commented on EventsExpress/ClientApp/src/config.js:
```diff
@@ -1,4 +1,7 @@
   "FACEBOOK_CLIENT_ID": "2741444662848625",
-  "GOOGLE_CLIENT_ID": "3429495463-t3uma25030ea0q819p2f600bi9j1869l.apps.googleusercontent.com"
+  "GOOGLE_CLIENT_ID": "3429495463-t3uma25030ea0q819p2f600bi9j1869l.apps.googleusercontent.com",
+  "TWITTER_CALLBACK_URL": "https://localhost:5002/Authentication/TwitterLogin",
```
This will not work properly on stage and prod, since it will not be redirected to localhost there. Consider using relative URL here.

iuriikhrystiuk commented on EventsExpress/ClientApp/src/services/EventsExpressService.js:
```diff
@@ -130,6 +131,13 @@ export default class EventsExpressService {
             : await res.json();
     }
 
+    setTwitterLoging = async (data) => {
```
Please use `setTwitterLogin` not  `setTwitterLogin`*g*

andrii-zaharovskyi replied:
Typo "g" fixed.

andrii-zaharovskyi committed:
The method setTwitterLoging has been renamed, also the client configuration file has been changed.

iuriikhrystiuk commented on EventsExpress/ClientApp/src/config.js:
```diff
@@ -1,4 +1,6 @@
-  "FACEBOOK_CLIENT_ID": "2741444662848625",
-  "GOOGLE_CLIENT_ID": "3429495463-t3uma25030ea0q819p2f600bi9j1869l.apps.googleusercontent.com"
+  "GOOGLE_CLIENT_ID": "3429495463-t3uma25030ea0q819p2f600bi9j1869l.apps.googleusercontent.com",
+  "TWITTER_CALLBACK_URL": "https://eventsexpress-test.azurewebsites.net/Authentication/TwitterLogin",
```
Do these changes work locally? Can you not use the relative url here?

andrii-zaharovskyi replied:
Reworked using the relative URL.

andrii-zaharovskyi committed:
The callback URL for the TwitterLogin has been changed.

andrii-zaharovskyi committed:
Facebook external login has been reverted and fixeed

commit 69c294ca3646373533cd92f9b92857e471a2839f
Date:   Tue Oct 6 13:13:13 2020 +0300

    Clean up

commit e292f27e8a4ff39bc74daf99afc8dbf11bb0e322
Date:   Tue Oct 6 13:01:24 2020 +0300

    Client App packge.json

commit 036932590d24ac25c34b07f5c5dc64c9159d5eb3
Date:   Tue Oct 6 12:06:58 2020 +0300

    The Facebook provider has been reverted and refactored

    ShareButton.js and FacebookLogin.js and login.js have been changed. Everything else has been formatted.

commit 83f3e2322ac7a029953b3bfee8e1f6af7e72fa5e
Date:   Mon Oct 5 21:20:01 2020 +0300

    Readded Facebook login

andrii-zaharovskyi committed:
Merge from development

commit 22a69ee732586ef7755052455254a93b451933ef
Merge: a748f93 06d15df
Author: Yevhen Vasylynchuk <45979893+Avalon412@users.noreply.github.com>
Date:   Mon Oct 5 23:37:16 2020 +0300

    Merge pull request #109 from EventsExpress-SoftServe/103-add-code-quality-analyzers

    103 add code quality analyzers

commit 06d15df37f84fd491e597878dbb3f9bb2b17d193
Author: Женя Василинчук <bensonatg412@gmail.com>
Date:   Mon Oct 5 16:51:51 2020 +0300

    Reconfigured JSHint to be runned with each build

commit ba3c901e74a875548fd373fabcd7d14a619cba7e
Author: Женя Василинчук <bensonatg412@gmail.com>
Date:   Mon Oct 5 16:35:33 2020 +0300

    Added and configured JSHint

commit a2156548927e3c96086e851929967ffa0003e474
Author: Женя Василинчук <bensonatg412@gmail.com>
Date:   Fri Oct 2 16:36:53 2020 +0300

    Added build config file and StyleCop Analyzer

commit a748f93b483e0f739c4a4e974be3fa7c9bdec6e8
Merge: bf7ad2f 6d92079
Author: Andrii Zaharovskyi <andrew.zaharovskyi@gmail.com>
Date:   Wed Sep 30 17:13:43 2020 +0300

    Merge pull request #105 from EventsExpress-SoftServe/sand0-patch-1

    Delete EventsExpress.xml

commit 6d92079c2ef419a4a81d453bf48ab0f42502a82a
Author: Alexandr Savchuk <44262349+sand0@users.noreply.github.com>
Date:   Wed Sep 30 17:08:24 2020 +0300

    Delete EventsExpress.xml

commit bf7ad2f4a2173022e87d69137ff05325a77b257e
Merge: 9a5beac da74169
Author: Yevhen Vasylynchuk <45979893+Avalon412@users.noreply.github.com>
Date:   Wed Sep 30 14:46:53 2020 +0300

    Merge pull request #104 from EventsExpress-SoftServe/99-fix-CommunaTimeAccoutning

    Fixed time accounting in 'Comunna'

commit da7416931aa28fe917ede872af447c428834ca81
Merge: a4f5073 9a5beac
Author: Yevhen Vasylynchuk <45979893+Avalon412@users.noreply.github.com>
Date:   Tue Sep 29 18:43:58 2020 +0300

    Merge branch 'development' into 99-fix-CommunaTimeAccoutning

commit a4f50734bca8a087fddf29057d23b397181f5c3e
Author: Женя Василинчук <bensonatg412@gmail.com>
Date:   Tue Sep 29 18:31:54 2020 +0300

    Fixed time accounting in 'Comunna'

andrii-zaharovskyi committed:
Merge branch 'development' into 83-integrate-twitter-auth

iuriikhrystiuk commented on EventsExpress/EventsExpress.csproj:
```diff
@@ -82,7 +82,7 @@
     <ItemGroup>
       <DistFiles Include="$(SpaRoot)build\**; $(SpaRoot)build-ssr\**" />
-      <ResolvedFileToPublish Include="@(DistFiles->'%(FullPath)')" Exclude="@(ResolvedFileToPublish)">
+      <ResolvedFileToPublish Include="@(DistFiles-&gt;'%(FullPath)')" Exclude="@(ResolvedFileToPublish)">
```
Why is this change required?

andrii-zaharovskyi replied:
My IDE makes this change automatically. This has been fixed.

andrii-zaharovskyi committed:
Using the version of the react-twitter-login library has been fixed + Clean up

iuriikhrystiuk commented on EventsExpress/ClientApp/.vscode/launch.json:
```diff
@@ -0,0 +1,18 @@
+{
```
Please modify `.gitignore` to ignore .vscode folders and do not iclude such in sources.

andrii-zaharovskyi replied:
Accidentally added `VSCode` files have been removed and the `.gitignore` file has been changed.

iuriikhrystiuk commented on EventsExpress/ClientApp/package-lock.json:
```diff
@@ -5520,6 +5520,11 @@
         "randomfill": "^1.0.3"
```
Please remove this file from PR.

andrii-zaharovskyi replied:
The file has been removed.

iuriikhrystiuk commented on EventsExpress/EventsExpress.xml:
```diff
@@ -15,21 +15,29 @@
         </member>
```
Please remove this file from PR.

andrii-zaharovskyi replied:
The file has been removed.

andrii-zaharovskyi committed:
Hot fixes
